src/NHibernate/Dialect/Function/RoundFunction.cs

```diff
@@ -24,9 +35,9 @@ public class RoundEmulatingSingleParameterFunction : ISQLFunction
 
 		public SqlString Render(IList args, ISessionFactoryImplementor factory)
 		{
```

Comment:
Either I do not get what you are thinking about, or it will be slightly more CPU lightweight but with a lot more code, rewriting a `SqlString` generation just for this case. For the later I do not think it is worth it.

```c#
var buf = new SqlStringBuilder();
buf.Add(_name)
   .Add("(");
for (int i = 0; i < args.Count; i++)
{
	object arg = args[i];
	if (arg is Parameter || arg is SqlString)
	{
		buf.AddObject(arg);
	}
	else
	{
		buf.Add(arg.ToString());
	}
	if (i < (args.Count - 1)) buf.Add(", ");
}

if (args.Count == 1)
	buf.Add(", 0");
return buf.Add(")").ToSqlString();
```

And this is an almost duplicate of `StandardSQLFunction`.

Reply:
Why would you need a loop here? You have 2 arguments. 

```
return new SqlString(
  _name,
  "(",
  args[0],
  ", ",
  args.Count > 1 ? args[1] : "0",
  ")"
)
```

Reply:
This would be breaking for dialects supporting more than two arguments (SQL-Server `round` case).